Objects/typeobject.c

```diff
@@ -6772,6 +6759,31 @@ type_ready_set_new(PyTypeObject *type)
     return 0;
 }
 
+static int
+type_ready_managed_dict(PyTypeObject *type)
+{
+    if (!(type->tp_flags & Py_TPFLAGS_MANAGED_DICT)) {
```

Comment:
Note to self: this flag is already set in one of the earlier `type_ready` checks.

Reply:
The flag may not be set at all. ``PyType_Ready`` is used for static types and types without managed dicts, too.